### PR TITLE
#262 Added user & password for ClickHouse unit tests

### LINKED ISSUE #262: Allow tests to use ClickHouse user & password
Current implementation does not allow to use authentication in unit tests

Adding the following properties would be a plus:

- CLICK_HOUSE_SERVER_USER
- CLICK_HOUSE_SERVER_PASSWORD

### PR DESCRIPTION
**Pull request by sbouchex**
Added user & password for ClickHouse unit tests

Close #262

### CODE REVIEW

File: clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java [hunk 1]
```diff
@@ -49,6 +51,12 @@ protected String getJdbcUrl(Object... params) {
             }
             sb.append(params[i]).append("=").append(params[i + 1]);
         }
+        if (SERVER_USER != null && SERVER_PASSWORD != null) {
```
Comment (sundy-li) [inline clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java:54]: how about 
```
if (SERVER_USER != null) 
....

and 

if (SERVER_PASSWORD != null)
....
```

File: clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java [hunk 2]
```diff
@@ -30,6 +30,8 @@
     protected static final ZoneId SERVER_TZ = ZoneId.of("UTC");
     protected static final String DRIVER_CLASS_NAME = "com.github.housepower.jdbc.ClickHouseDriver";
     protected static final int SERVER_PORT = Integer.parseInt(System.getProperty("CLICK_HOUSE_SERVER_PORT", "9000"));
+    protected static final String SERVER_USER = System.getProperty("CLICK_HOUSE_SERVER_USER", null);
```
Comment (sundy-li) [inline clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java:33]: default user should be 'default' instead of null.

Comment (sundy-li) [reply clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java:33]: default password can be empty string.

Comment (sbouchex) [reply clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java:33]: I'll change that

Comment (sundy-li) [reply clickhouse-native-jdbc/src/test/java/com/github/housepower/jdbc/AbstractITest.java]: the default password can be empty string too,   ""

**Comment by sundy-li** [comment]
@sbouchex  try to rebase master, it should be ok.

**Comment by sbouchex** [comment]
@sundy-li Thanks for the changes but I don't need to do anything else. Changes must be merge to the master branches on your side now.

**Comment by pan3793** [comment]
@sbouchex @sundy-li Merry Christmas and thanks for work on it.

Manually rebased on master and run test in local, there a one UT broken, and it was fixed in https://github.com/housepower/ClickHouse-Native-JDBC/commit/31d27604acf9a5098ffa3ed2c465f382fe3fec49

With this patch, all UTs and ITs passed. I will merge it.

> but I don't need to do anything else

@sbouchex `git rebase` is a more powerful command than `git merge`, there are some gifs illustrate some commands of `git`  https://dev.to/lydiahallie/cs-visualized-useful-git-commands-37p1

**Pull request merged**